### ISSUE TITLE
Support bazel go_repository urls field

### ISSUE THREAD

**Issue by uri-canva** [renovatebot/renovate#6207]
**What would you like Renovate to be able to do?**
Use renovate to update `go_repository` rules that use the `urls` field.

**Describe the solution you'd like**
Specify `urls` like `https://codeload.github.com/pkg/errors/zip/v0.9.1`, `https://codeload.github.com/google/go-containerregistry/zip/221517453cf931400e6607315045445644122692`, and similar and have renovate update them based on the extracted tag or commit.

**Describe alternatives you've considered**
Using `tag` or `commit` which are supported interferes with bazel's caching mechanism, using `sum` isn't implemented yet (though it has been requested in https://github.com/renovatebot/renovate/issues/4402) and causes different repositories using gazelle to interact with each other when you wouldn't want them to.

**Additional context**
See https://github.com/bazelbuild/rules_docker/pull/1452 for an example of why one would prefer the `urls` field over `tag`, `commit` or `sum`.

**Comment by viceice** [comment]
Looks like we need to compute the sha256 of the zip file?

**Comment by rarkins** [comment]
I started working on moving bazel to autoUpdate. Any sha256 we need to calculate can then be done as part of artifacts so that we're not calculating them every run.

**Comment by viceice** [comment]
Can we use digest update for the hash here? So we even don't need artifacts

**Comment by rarkins** [comment]
No, we need artifacts to do Bazel properly anyway. And we don't want to be *calculating* the sha256 every time.

**Comment by samorm** [comment]
pkg install python

**Comment by rarkins** [comment]
Blocked by #9667
